### ISSUE TITLE
just in case

### ISSUE THREAD

**Issue by TrejGun** [rackt/react-router#2738]
i have checked a how many tickets you have closed with test like 

```
Thanks for your question but we don't care...
```

guys, shame on you, nobody treats their community like you do! 87 developers were sent to look for a help somewhere else, and will never come back again, and hopefully won't use your lib. Thanks for your advise but fortunately i'm not banned by google and stackoverflow; and can search and ask. And i'm asking my questions in place, i think, is most right for this. Check how many replies there are on stackoverflow - https://stackoverflow.com/questions/tagged/react-router the average amount of answers is close to zero. 

Guess why do you have so many questions? because you docs sucks! its completely not obvious how to build react app after you were working with angular.

**Comment by knowbody** [comment]
Hi, I appreciate your feedback. But before you start judging all of us, you should learn what were the reasons why we ask people to direct their questions to StackOverflow or Discord. If you've ever managed a large open source project you would probably know that you can't make everyone happy. We try very hard to make this library as great as possible.

All of us (maintainers) have their own jobs, lives and the work on React Router is mostly something what we do in our free non-paid time.

Comments like:
> guys, shame on you, nobody treats their community like you do! 87 developers were sent to look for a help somewhere else, and will never come back again, and hopefully won't use your lib.

do not motivate us, BUT on the other hand I know how many people appreciate the work we do here, and those are the people we do that for.

> Guess why do you have so many questions? because you docs sucks! its completely not obvious how to build react app after you were working with angular.

I assume people who come to use the library built on top of react should at least know basics, we are not here to teach you how to program or hold your hand and give you a copy-paste answer.

Talking about StackOverflow, I try to spend 30-40 mins every day and answer as many questions with `react-router` tag as I can.

We know that the docs are not the best for the beginners. @ryanflorence works on a great tutorial for react router. It's only a few of us who maintain this repo, and is hard to ship new features, write great docs, fix bugs and answer questions.

Open source is about community and if you see that something is broken, not good enough, please send a PR with suggested changes that will help us and benefit others.

**Comment by TrejGun** [comment]
@knowbody 

sorry for `hopefully won't use your lib` this was not motivating at all. and yes i appreciate your effort to deliver a good component

yesterday i opened an issue #2737 where i'm not asking how to code js but about a best way to use your lib. I checked what your docs says about redirecting - sad story, it says nothing

[API](https://github.com/rackt/react-router/blob/latest/docs/API.md#redirect)
[RouteConfiguration](https://github.com/rackt/react-router/blob/latest/docs/guides/basics/RouteConfiguration.md#preserving-urls)
[RouteMatching](https://github.com/rackt/react-router/blob/latest/docs/guides/basics/RouteMatching.md#precedence)
[CHANGES](https://github.com/rackt/react-router/blob/latest/CHANGES.md#redirect-route)

the half of examples in google and stackoverflow are related v0.13, and you are going to change everything again in v2. So no docs, no examples, not stable api, where should i start? i have to read source code, and may be fix it and do PR. And may be i want to do so but who knows where you are moving your project, remember that road map is only in your head and you don't want to communicate to share your vision.

**Comment by gaearon** [comment]
>yesterday i opened an issue #2737 where i'm not asking how to code js but about a best way to use your lib.

You received a reply: issues are not the best way to ask questions. What is the problem with asking the same question on StackOverflow? Ask it there and if you don't receive a response, ping people on Discord with that question link. All we're saying is that we don't use issues for questions because they get lost easily and people tend to ask the same things over and over again. StackOverflow is a better place for questions, this is why your issue was closed.

>I checked what your docs says about redirecting - sad story, it says nothing

I don't understand your point. You just linked to four pages that explicitly deal with redirecting. How is this “nothing”?

>So no docs

You just linked to them. What do you mean by “no docs”?

>no examples

[Is 15 examples in “examples” folder not enough?](https://github.com/rackt/react-router/tree/master/examples)

>And may be i want to do so but who knows where you are moving your project, remember that road map is only in your head and you don't want to communicate to share your vision.

Um, no, it’s in an issue called “v2 Roadmap” that is right on the first page of issues: https://github.com/rackt/react-router/issues/2646.

I'm surprised by the constant use of hyperbole in your comments. Docs and examples aren't in ideal shape but claiming they don't exist is absurd.

**Comment by TrejGun** [comment]
@gaearon so you dont have a time to answer my question, but have time to write all this? nice...

i spend whole day trying to understand how this lib works
and still have time to answer you 

> You just linked to them. What do you mean by “no docs”?

it means docs says nothing about how, for example, to do a redirect to another site
and this is not obvious that i should do route and use a hook
```
<Route path="*" component={MyComponent}
	onEnter={(nextState) => {
		window.location.href = "http://google.com?q="+nextState.params.splat;
	}}
/>
```

it should be something simple like 
```
<Redirect from="*" to="http://google.com?q=*" />
/>
```

**Comment by timdorr** [comment]
If you want to redirect to another site, that's not really the job of react-router. Use an `a` tag or `location.href`. We can't manage routes to things outside of our scope.

**Comment by taion** [comment]
This discussion is no longer productive, and this isn't the right place to ask questions regarding how to set up redirects. I'm locking this issue for the time being.

**Comment by ryanflorence** [comment]
@TrejGun is right, getting started with react router is pretty rough right now since we've focused primarily on the code and getting it right.

As soon as the next release is out I've got a really great tutorial/getting started guide that will cover all the primary use-cases so you don't have to go hunting through API docs to figure anything out (there are very many ways to redirect, if you're new to the router then the current docs would be a nightmare to figure out)

**Also, contributors, listen up**: The router has crossed the line that divides those that willfully use our stuff and those that simply have to because its what their project at work is using. They've got a task they have to complete and when stuff is confusing its super frustrating. They aren't trying to help us build this thing and that's fine, they're just doing their job.

We're going to get more of these kinds of comments, just put on a filter, find the nugget of truth wrapped up in the vitriolic fruits of frustration, and then improve the docs accordingly :)
